k3s-io/k3s
# Only clean up containerd hosts dirs managed by k3s

#### Proposed Changes ####

Only clean up containerd hosts dirs managed by k3s

If the user creates their own hosts.toml, or drops CA or client certs in the directory by hand, we shouldn't delete them whenever K3s starts.

#### Types of Changes ####

enhancement

#### Verification ####
1. Create your own host dir at `/var/lib/rancher/k3s/agent/etc/containerd/certs.d/MYREGISTRY` and add custom hosts.toml and/or certs. Restart K3s, and note that the custom content you created is left alone.
2. Add an entry for MYREGISTRY in registries.yaml and restart k3s, note that the custom files are cleaned up and replaced with the K3s-managed hosts.toml
3. Edit the file at `/var/lib/rancher/k3s/agent/etc/containerd/certs.d/MYREGISTRY/hosts.toml` to remove the DO NOT EDIT header and restart k3s; note that it is reverted to the managed content.
4. Remove MYREGISTRY from registries.yaml and restart k3s. Note that `/var/lib/rancher/k3s/agent/etc/containerd/certs.d/MYREGISTRY` is removed.

#### Testing ####


#### Linked Issues ####

* https://github.com/k3s-io/k3s/issues/10787

#### User-Facing Change ####

```release-note
```

#### Further Comments ####
